file_manipulation/convert_file.py

```diff
@@ -0,0 +1,269 @@
+        default='',
+        required=False)
+    parser.add_argument(
+        '--destination-folder-name',
+        dest='destination_folder_name',
+        default='',
+        required=False)
+    parser.add_argument(
+        '--destination-file-name',
+        dest='destination_file_name',
+        default=None,
+        required=False)
+    parser.add_argument('--destination-file-format', dest='destination_file_format', choices={
+        'tsv',
+        'psv',
+        'xlsx',
+        'parquet',
+        'stata',
+        'hdf5'}, required=True)
+    parser.add_argument('--extra-args', dest='extra_args', default='{}')
+    return parser.parse_args()
+
+
+def extract_file_name_from_source_full_path(source_full_path):
+    """
+    Use the file name provided in the source_full_path variable. Should be run only
+    if a destination_file_name is not provided.
+    """
+    destination_file_name = os.path.basename(source_full_path)
+    return destination_file_name
+
+
+def clean_folder_name(folder_name):
+    """
+    Cleans folders name by removing duplicate '/' as well as leading and trailing '/' characters.
+    """
+    folder_name = folder_name.strip('/')
+    if folder_name != '':
+        folder_name = os.path.normpath(folder_name)
+    return folder_name
+
+
+def combine_folder_and_file_name(folder_name, file_name):
+    """
+    Combine together the provided folder_name and file_name into one path variable.
+    """
+    combined_name = os.path.normpath(
+        f'{folder_name}{"/" if folder_name else ""}{file_name}')
+    combined_name = os.path.normpath(combined_name)
+
+    return combined_name
+
+
+def find_all_local_file_names(source_folder_name):
+    """
+    Returns a list of all files that exist in the current working directory,
+    filtered by source_folder_name if provided.
+    """
+    cwd = os.getcwd()
+    cwd_extension = os.path.normpath(f'{cwd}/{source_folder_name}/**')
+    file_names = glob.glob(cwd_extension, recursive=True)
+    return [file_name for file_name in file_names if os.path.isfile(file_name)]
+
+
+def find_all_file_matches(file_names, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for file in file_names:
+        if re.search(file_name_re, file):
+            matching_file_names.append(file)
+
+    return matching_file_names
+
+
+def create_fallback_destination_file_name(source_file_name, destination_file_format):
+    """
+    If a destination_file_name is not provided, uses the source_file_name with a removal of the compression extension.
+    """
+
+    format_extensions = {
+        "tsv": ".tsv",
+        "psv": ".psv",
+        "xlsx": ".xlsx",
+        "parquet": ".parquet",
+        "stata": ".dta",
+        "hdf5": ".h5"
+    }
+
+    file_name = os.path.basename(source_file_name)
+    file_name = file_name.replace(
+        '.csv', f'{format_extensions[destination_file_format]}')
```

Comment:
This replace isn't the most generic logic that will work for all file names. 

A file name of `hello.csv.world.txt` would be replaced with `hello.xlsx.world.txt` for example. You need an algorithm that replaces just the extension, or suffix, of the strings. 

Quick Google search provides this: https://stackoverflow.com/questions/3548673/how-to-replace-or-strip-an-extension-from-a-filename-in-python